boto/endpoints.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import logging
+
+from boto.vendored.regions.regions import EndpointResolver
+
+
+# Since we will be resolving every region, it's worth not cluttering up the
+# logs with all that data.
+_endpoint_logger = logging.getLogger('boto.vendored.regions.regions')
+_endpoint_logger.disabled = True
+
+
+class BotoEndpointResolver(EndpointResolver):
+    """Endpoint resolver which handles boto2 compatibility concerns."""
+
+    SERVICE_RENAMES = {
+        # The botocore resolver is based on endpoint prefix.
+        # These don't always sync up to the name that boto2 uses.
+        # A mapping can be provided that handles the mapping between
+        # "service names" and endpoint prefixes.
+        'awslambda': 'lambda',
+        'cloudwatch': 'monitoring',
+        'ses': 'email',
+        'ec2containerservice': 'ecs',
+        'configservice': 'config',
+    }
+
+    def __init__(self, endpoint_data, service_rename_map=None):
```

Comment:
No. Actually, I should probably make that default private. We need both the boto2 -> endpoint_prefix mapping and the endpoint_prefix -> boto2 mapping. To avoid reversing the mapping every time it's necessary, it is reversed when the class is constructed.